report-portfolio/portfolio.Rmd

```diff
@@ -0,0 +1,234 @@
+---
+title: "Portolio Report"
+params:
+  date:
+    input: date
+    label: Start Date
+    value: '2010-01-01'
+  mar:
+    input: slider
+    label: Min Acceptable Rate
+    max: 0.5
+    min: 0
+    step: 0.001
+    value: 0.008
+  portfolio:
+    choices:
+    - balanced_portfolio_returns
+    - aggressive_portfolio_returns
+    - conservative_portfolio_returns
+    input: select
+    label: portfolio
+    value: balanced_portfolio_returns
+  portfolioName:
+    input: text
+    label: title
+    value: Balanced
+  window:
+    input: numeric
+    label: Rolling Window
+    max: 36
+    min: 6
+    value: 12
+resource_files:
+- config.yml
+output:
+  flexdashboard::flex_dashboard:
+    orientation: rows
+    source_code: embed
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = FALSE, message = FALSE, warning = FALSE)
+
+library(config)
+library(dbplyr)
+library(openxlsx)
+library(tidyverse)
+library(tidyquant)
+library(highcharter)
+library(timetk)
+library(blastula)
+library(formattable)
+library(DBI)
+library(RSQLite)
+
+db_path <- config::get("db", file = "config.yml")
+con <- DBI::dbConnect(RSQLite::SQLite(), db = db_path)
+```
+
+```{r}
+balanced_portfolio_returns <- tbl(con, "balanced_portfolio_returns")
+conservative_portfolio_returns <- tbl(con, "conservative_portfolio_returns")
+aggressive_portfolio_returns <- tbl(con, "aggressive_portfolio_returns")
+```
+
+```{r}
+MAR <- params$mar
+# run our calcs
+portfolio_selected <- 
+    eval(as.name(params$portfolio)) %>%
```

Comment:
I don't like this and the code chunk above very much ... why not just:

```
portfolio_selected <- tbl(con, params$portfolio)
```

Reply:
Ahhh. Much better!